src/components/views/rooms/RoomPreviewBar.js

```diff
@@ -122,8 +139,31 @@ module.exports = React.createClass({
                 </div>
             );
 
-        }
-        else if (this.props.error) {
+        } else if (kicked || banned) {
+            const verb = kicked ? 'kicked' : 'banned';
+            const roomName = this._roomNameElement('this room');
+            const kicker = MatrixClientPeg.get().getUser(
+                myMember.events.member.getSender()
+            );
+            let reason;
+            if (myMember.events.member.getContent().reason) {
+                reason = <div>Reason: {myMember.events.member.getContent().reason}</div>
+            }
+            let rejoinBlock;
+            if (!banned) {
+                rejoinBlock = <a onClick={ this.props.onJoinClick }><b>Rejoin</b></a>;
+            }
+            joinBlock = (
+                <div>
+                    <div className="mx_RoomPreviewBar_join_text">
+                        You have been {verb} from {roomName} by {kicker.displayName}.<br />
```

Comment:
As per RL: There are no guarantees that the person who banned you will exist when you `getMember` (e.g. invite -> ban).